index.bs

```diff
@@ -4115,6 +4147,811 @@ opened</dfn> steps given |window|, |type| and |message|.
 
 </div>
 
+## The network Module ## {#module-network}
+
+The <dfn export for=modules>network</dfn> module contains commands and events
+relating to network requests.
+
+### Definition ### {#module-network-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+NetworkCommand = (
+)
+
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+NetworkResult = (
+)
+
+NetworkEvent = (
+    network.BeforeRequestSent //
+    network.FetchError //
+    network.ResponseStarted //
+    network.ResponseCompleted
+)
+
+</pre>
+
+A [=remote end=] has a <dfn>before request sent map</dfn> which is initially an
+empty map.  It's used to track the network events for which a
+<code>network.beforeRequestSent</code> event has already been sent.
+
+### Types ### {#module-network-types}
+
+#### The network.BaseParameters type #### {#type-network-BaseParameters}
+
+<pre class="cddl local-cddl">
+network.BaseParameters = {
+    context: BrowsingContext / null,
+    isRedirect: bool,
+    navigation: Navigation / null,
+    redirectCount: js-uint,
+    request: network.RequestData,
```

Comment:
`BaseParameters` always contains `request`. Does it make sense in case of response events like `network.responseStarted` or `network.responseCompleted`?

Reply:
Yes, although other designs are possible.

The constraint is that we at least need to provide the request id, so you know which response this is a request for.

In the current design we also provide the timing data in the `RequestData` since it applies to both request and response phases.

The tradeoff of the current setup is that if you subscribe to more than one event you get duplicate data. But it also means you aren't required to subscribe to all events e.g. it would be possible to just subscribe to `network.responseCompleted` to find out which requests occurred.

Reply:
Sounds reasonable.